src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -855,14 +856,19 @@ function buildInvestmentStats(
 
 function buildObjectives(
   item: DestinyItemComponent,
+  owner: D2Store | undefined,
   objectivesMap: { [key: string]: DestinyItemObjectivesComponent },
   objectiveDefs: LazyDefinition<DestinyObjectiveDefinition>
 ): DimObjective[] | null {
   if (!item.itemInstanceId || !objectivesMap[item.itemInstanceId]) {
     return null;
   }
 
-  const objectives = objectivesMap[item.itemInstanceId].objectives;
+  const characterProgressionObjectives =
+    owner && owner.progression ? owner.progression.progressions[item.itemHash] : [];
```

Comment:
Actually looking at this again, I’m not sure it’s correct. Can you check with the new Truth quest? I think you need to be referencing a property called uninstancedItemObjectives. This code you’ve written is indexing the character progressions, which are a different thing (they’re indexed by progression hash, not item hash)